jax/lax/lax.py

```diff
@@ -1188,13 +1188,13 @@ def cumprod(operand: Array, axis: int) -> Array:
   """Computes a cumulative product along `axis`."""
   return cumprod_p.bind(operand, axis=int(axis))
 
-def sort(operand: Union[Array, Tuple[Array, ...]], dimension: int = -1
+def sort(operand: Union[Array, Sequence[Array]], dimension: int = -1
          ) -> Union[Array, Tuple[Array, ...]]:
   """Wraps XLA's `Sort
   <https://www.tensorflow.org/xla/operation_semantics#sort>`_
   operator.
   """
-  if isinstance(operand, tuple):
+  if isinstance(operand, Sequence):
```

Comment:
Is this `typing.Sequence` or `collections.abc.Sequence`?

Only the later will work for runtime checks, but I might be inclined to be even more restrictive and use `(list, tuple)` instead.

Reply:
This is `typing.Sequence`, which has the semantics we would like in this case:
```python
>>> import numpy as np
>>> import jax.numpy as jnp
>>> from typing import Sequence
>>> isinstance([], Sequence)
True
>>> isinstance((), Sequence)
True
>>> isinstance(np.arange(4), Sequence)
False
>>> isinstance(jnp.arange(4), Sequence)
False
```

Reply:
Although it is a bit overly permissive:
```python
>>> isinstance('abc', Sequence)
True
```
but in this case, the resulting error will be the same whether we check for `Sequence` or check for `(list, tuple)`.